x-pack/plugins/alerting/server/rules_client/lib/siem_legacy_actions/format_legacy_actions.ts

```diff
@@ -0,0 +1,145 @@
+import pMap from 'p-map';
+import { RuleAction, Rule } from '../../../types';
+import type { RuleExecutorServices } from '../../..';
+import { injectReferencesIntoActions } from '../../common';
+import { transformToNotifyWhen } from './transform_to_notify_when';
+import { transformFromLegacyActions } from './transform_legacy_actions';
+import { LegacyIRuleActionsAttributes, legacyRuleActionsSavedObjectType } from './types';
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ */
+interface LegacyGetBulkRuleActionsSavedObject {
+  alertIds: string[];
+  savedObjectsClient: RuleExecutorServices['savedObjectsClient'];
+  logger: Logger;
+}
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ */
+export interface LegacyActionsObj {
+  ruleThrottle: string | null;
+  legacyRuleActions: RuleAction[];
+}
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ * this function finds all legacy actions associated with rules in bulk
+ * it's useful for such methods as find, so we do not request legacy actions in a separate request per rule
+ * @params params.alertIds - list of rule ids to look for legacy actions for
+ * @params params.savedObjectsClient - savedObjectsClient
+ * @params params.logger - logger
+ * @returns map of legacy actions objects per rule with legacy actions
+ */
+export const legacyGetBulkRuleActionsSavedObject = async ({
+  alertIds,
+  savedObjectsClient,
+  logger,
+}: LegacyGetBulkRuleActionsSavedObject): Promise<Record<string, LegacyActionsObj>> => {
+  const references = alertIds.map<SavedObjectsFindOptionsReference>((alertId) => ({
+    id: alertId,
+    type: 'alert',
+  }));
+  const errors: unknown[] = [];
+  const results = await pMap(
+    chunk(references, 1000),
+    async (referencesChunk) => {
+      try {
+        return savedObjectsClient.find<LegacyIRuleActionsAttributes>({
+          // here we looking legacyRuleActionsSavedObjectType, as not all of rules create `siem.notifications`
+          // more information on that can be found in https://github.com/elastic/kibana/pull/130511 PR
+          type: legacyRuleActionsSavedObjectType,
+          perPage: 10000,
+          hasReference: referencesChunk,
+        });
+      } catch (error) {
+        logger.error(
+          `Error fetching rule actions: ${error instanceof Error ? error.message : String(error)}`
+        );
+        errors.push(error);
+        return [];
+      }
+    },
+    { concurrency: 1 }
+  );
+  const actionSavedObjects = results.flat().flatMap((r) => r.saved_objects);
+
+  if (errors.length) {
+    throw new AggregateError(errors, 'Error fetching rule actions');
```

Comment:
let's mute the error, we should log it for sure!

Reply:
added try/catch in `formatLegacyActions` and `migrateLegacyActions` on top level. So any thrown errors will be caught and logged